components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/DefaultApplicationValidatorTest.java

```diff
@@ -172,4 +182,65 @@ public void validateAdaptiveAuthScriptTest(String isValidationFailScenario, Stri
                     String.join("|", validationErrors));
         }
     }
+
+    @DataProvider(name = "validateTrustedAppMetadataDataProvider")
+    public Object[][] validateTrustedAppMetadataDataProvider() {
+
+        String errorThumbprints = String.join(",", Collections.nCopies(21, "thumbprint"));
+        return new Object[][]{
+                {"com.wso2.sample", "thumbprint1,thumbprint2", "sample.app.id", false, false, false},
```

Comment:
Since there is about 5 validation that we do shall we add a comment in each of the data lines which validation it covers.